### PR TITLE
feature/add-milesights

### PR DESCRIPTION
**Pull request by J2thatsme**

### CODE REVIEW

File: types/milesight/WS202/uplink.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    const channelId = bytes[i++];
+    const channelType = bytes[i++];
+    // BATTERY
+    if (channelId === 0x01 && channelType === 0x75) {
+      lifecycle.batteryLevel = bytes[i];
+      i += 1;
+    }
+    // PIR
+    else if (channelId === 0x03 && channelType === 0x00) {
+      data.pir = Number(bytes[i]);
```
Comment (Biglr) [inline types/milesight/WS202/uplink.js:28]: not sure this is gonna work as intended :thinking:

**Review by Biglr — CHANGES_REQUESTED** [review]

File: types/milesight/WS202/uplink.js (hunk 1 above)
Comment (J2thatsme) [reply types/milesight/WS202/uplink.js:28]: Seems to work, would you suggest to do this without type casting ?

Comment (J2thatsme) [reply types/milesight/WS202/uplink.js:28]: The bytes array contains f.e the following values:  [
  1,
  117,
  16,
  3,
  0,
  1,
  4,
  0,
  0,
]
It's one byte per entry but not encoded as binary.

Comment (Biglr) [reply types/milesight/WS202/uplink.js:28]: ok, so the values are already parsed from hex to numbers? make sure they are numbers and use parseInt() (or double or whatever) if you want to convert something to numbers ^^

Comment (J2thatsme) [reply types/milesight/WS202/uplink.js:28]: 👍 they get parsed automatically but the decoder expects this.

**Review by Biglr — APPROVED** [review]

**Pull request merged**